homeassistant/components/nest/climate.py

```diff
@@ -192,8 +192,7 @@ def current_temperature(self):
     def hvac_mode(self):
         """Return current operation ie. heat, cool, idle."""
         if self._mode == NEST_MODE_ECO:
-            # We assume the first operation in operation list is the main one
-            return self._operation_list[0]
+            return self.device.previous_mode
```

Comment:
We're only allowed to return the modes available in the base climate component. That's why we look up a dictionary that maps between nest and hass modes below.